fuel-vm/src/interpreter/executors/instruction.rs

```diff
@@ -303,31 +304,67 @@ where
             Instruction::JI(ji) => {
                 self.gas_charge(self.gas_costs.ji)?;
                 let imm = ji.unpack();
-                self.jump(imm.into())?;
+                self.jump(true, JumpMode::Absolute, imm.into(), 0)?;
             }
 
             Instruction::JNEI(jnei) => {
                 self.gas_charge(self.gas_costs.jnei)?;
                 let (a, b, imm) = jnei.unpack();
-                self.jump_not_equal(r!(a), r!(b), imm.into())?;
+                self.jump(r!(a) != r!(b), JumpMode::Absolute, imm.into(), 0)?;
             }
 
             Instruction::JNZI(jnzi) => {
                 self.gas_charge(self.gas_costs.jnzi)?;
                 let (a, imm) = jnzi.unpack();
-                self.jump_not_zero(r!(a), imm.into())?;
+                self.jump(r!(a) != 0, JumpMode::Absolute, imm.into(), 0)?;
             }
 
             Instruction::JMP(jmp) => {
                 self.gas_charge(self.gas_costs.jmp)?;
                 let a = jmp.unpack();
-                self.jump(r!(a))?;
+                self.jump(true, JumpMode::Absolute, r!(a), 0)?;
             }
 
             Instruction::JNE(jne) => {
                 self.gas_charge(self.gas_costs.jne)?;
                 let (a, b, c) = jne.unpack();
-                self.jump_not_equal(r!(a), r!(b), r!(c))?;
+                self.jump(r!(a) != r!(b), JumpMode::Absolute, r!(c), 0)?;
+            }
+
+            Instruction::JMPF(jmpf) => {
+                self.gas_charge(self.gas_costs.jmpf)?;
+                let (a, offset) = jmpf.unpack();
+                self.jump(true, JumpMode::RelativeForwards, r!(a), offset.into())?;
```

Comment:
`offset` could be part of the `JumpMode`

Reply:
So could `addr`, but all modes share it, so then we would create `struct JumpArgs { addr, offset, mode }`. I don't feel like this makes much sense.

Reply:
But it will remove passing `0` for `Absolute` mode=) The `addr` is used in all modes, so we don't need to move it into 
`JumpMode`

Reply:
I rewrote this to use a builder pattern, that seems cleaner